cloud/test/integration/fixtures/devicemodel.go

```diff
@@ -0,0 +1,401 @@
+			BluetoothOperationType:  operationType,
+			BluetoothOperationValue: value,
+		}
+		op.devicePropertyVisitor.VisitorConfig.Bluetooth.BluetoothDataConverter.OrderOfOperations =
+			append(op.devicePropertyVisitor.VisitorConfig.Bluetooth.BluetoothDataConverter.OrderOfOperations, bluetoothOperation)
+	}
+
+}
+
+func withRegister(register v1alpha1.ModbusRegisterType) DevicePropertyVisitorOption {
+	return func(op *DevicePropertyVisitorOp) {
+		op.devicePropertyVisitor.VisitorConfig.Modbus.Register = register
+	}
+}
+
+func withOffset(offset int64) DevicePropertyVisitorOption {
+	return func(op *DevicePropertyVisitorOp) {
+		op.devicePropertyVisitor.VisitorConfig.Modbus.Offset = offset
+	}
+}
+
+func withLimit(limit int64) DevicePropertyVisitorOption {
+	return func(op *DevicePropertyVisitorOp) {
+		op.devicePropertyVisitor.VisitorConfig.Modbus.Limit = limit
+	}
+}
+
+func withNodeID(nodeID string) DevicePropertyVisitorOption {
+	return func(op *DevicePropertyVisitorOp) {
+		op.devicePropertyVisitor.VisitorConfig.OpcUA.NodeID = nodeID
+	}
+}
+
+func withBrowseName(browseName string) DevicePropertyVisitorOption {
+	return func(op *DevicePropertyVisitorOp) {
+		op.devicePropertyVisitor.VisitorConfig.OpcUA.BrowseName = browseName
+	}
+}
+
+func (op *DevicePropertyVisitorOp) applyDevicePropVisitorOpts(opts []DevicePropertyVisitorOption) {
+	for _, opt := range opts {
+		opt(op)
+	}
+}
+
+func newDevicePropVisitorOp(opts ...DevicePropertyVisitorOption) *DevicePropertyVisitorOp {
+	op := &DevicePropertyVisitorOp{
+		devicePropertyVisitor: v1alpha1.DevicePropertyVisitor{},
+	}
+	op.applyDevicePropVisitorOpts(opts)
+	return op
+}
+
+func newDeviceModel(name string, namespace string) *v1alpha1.DeviceModel {
+	spec := v1alpha1.DeviceModelSpec{}
+	deviceModel := &v1alpha1.DeviceModel{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: apiVersion,
+			Kind:       deviceModelKind,
+		},
+		Spec: spec,
+	}
+	return deviceModel
+}
+
+func DeviceModelWithPropertyNoName(name string, namespace string) *v1alpha1.DeviceModel {
+	deviceModel := newDeviceModel(name, namespace)
+	devicePropertyOp := newDevicePropertyOp(withDescription(devicePropertyTemperatureDesc),
+		withStringType(v1alpha1.PropertyAccessMode(v1alpha1.ReadOnly), ""))
+	deviceModel.Spec.Properties = append(deviceModel.Spec.Properties, devicePropertyOp.deviceProperty)
+	return deviceModel
+}
+
+func DeviceModelWithPropertyNoType(name string, namespace string) *v1alpha1.DeviceModel {
+
+	deviceModel := newDeviceModel(name, namespace)
+	devicePropertyOp := newDevicePropertyOp(withName(devicePropertyTemperature), withDescription(devicePropertyTemperatureDesc))
+	deviceModel.Spec.Properties = append(deviceModel.Spec.Properties, devicePropertyOp.deviceProperty)
+	return deviceModel
+}
+
+func DeviceModelWithPropertyBadAccessMode(name string, namespace string) *v1alpha1.DeviceModel {
+
```

Comment:
could you pls remove those white spaces?

Reply:
done